src/Analysis/Ast/Impl/Dependencies/DependencyResolver.cs

```diff
@@ -129,12 +130,12 @@ public int ChangeValue(TKey key, TValue value, params TKey[] incomingKeys)
         }
 
         private static ImmutableArray<WalkingVertex<TKey, TValue>> CreateWalkingGraph(ImmutableArray<DependencyVertex<TKey, TValue>> snapshot, ImmutableArray<DependencyVertex<TKey, TValue>> changedVertices) {
-            var analysisGraph = ImmutableArray<WalkingVertex<TKey, TValue>>.Empty;
+            var analysisGraph = new List<WalkingVertex<TKey, TValue>>(changedVertices.Count);
```

Comment:
you can use ImmutableArray.Builder rather than use List and then do ToImmutableArray

Reply:
There is no builder; this is our own implementation in our repo, not the widely available one.

Reply:
I see. thank you for the answer. just curious. can I ask why you guys are not using the public immutable array? they also has ImmutableList that is baed on tree.

Reply:
`ImmutableList` is slow. Even for enumeration and accessing element by index. `ImmutableArray` is fast on reading but very slow on modification. We know that most of our operations are either adding or iterating, so I've added a type that is fast for that.

Reply:
by the way, since you guys are using .net core.

moving to new Span<T> - https://msdn.microsoft.com/en-us/magazine/mt814808.aspx - could be another way to improve perf around these?

Reply:
Sure, but it isn't part of .Net Standard yet. We will start using it as soon as .net standard 2.1 is released.